### PR TITLE
EASY-2450: custom license in header bij file download vanuit easy-download

### PR DESCRIPTION
**Pull request by vesaakerman**
Fixes EASY-2450

#### When applied it will
* remove fetching of `license` in this module
* get `license key` and `license title` from `authorization information` (easy-auth-info)
  (in authorization information `custom license` when applicable)

#### How should this be manually tested?
curl -v http://deasy.dans.knaw.nl:20160/ark:/73189/<bag-id\>/data/path/to/file
#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-auth-info                      | [PR#33](https://github.com/DANS-KNAW/easy-auth-info/pull/33)     | 
easy-solr4files                      | [PR#53](https://github.com/DANS-KNAW/easy-solr4files-index/pull/53)     |

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -148,33 +148,6 @@
...
-                        <phase>generate-resources</phase>
-                        <goals>
-                            <goal>unpack</goal>
-                        </goals>
-                        <configuration>
-                            <artifactItems>
-                                <artifactItem>
-                                    <groupId>nl.knaw.dans.easy</groupId>
-                                    <artifactId>easy-licenses</artifactId>
-                                    <version>${easy.licenses.version}</version>
```
Comment (rvanheest) [inline pom.xml]: please also remove the property in the top of `pom.xml`

Comment (rvanheest) [inline pom.xml]: please also remove the property in the top of `pom.xml`

Comment (rvanheest) [inline pom.xml]: please also remove the property in the top of `pom.xml`

File: src/main/scala/nl.knaw.dans.easy.download/components/FileItem.scala [hunk 2]
```diff
@@ -36,7 +36,9 @@ case class FileItem(itemId: String,
                     owner: String,
                     dateAvailable: DateTime,
                     accessibleTo: RightsFor.Value,
-                    visibleTo: RightsFor.Value
+                    visibleTo: RightsFor.Value,
+                    licenseKey: String,
+                    licenseTitle: String
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.download/components/FileItem.scala]:
```suggestion
                    licenseTitle: String,
```

File: src/test/scala/nl.knaw.dans.easy.download/components/FileItemSpec.scala [hunk 3]
```diff
@@ -30,31 +30,39 @@ class FileItemSpec extends TestSupportFixture {
     FileItem(itemId = "uuid/file.txt", owner = "someone",
       dateAvailable = new DateTime("4016-12-15"),
       accessibleTo = RESTRICTED_REQUEST,
-      visibleTo = RESTRICTED_REQUEST
+      visibleTo = RESTRICTED_REQUEST,
+      licenseKey = "http://opensource.org/licenses/MIT",
+      licenseTitle = "MIT.txt"
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.download/components/FileItemSpec.scala]: add those comma's in this file for every diff as well...

File: src/main/scala/nl.knaw.dans.easy.download/EasyDownloadServlet.scala [hunk 4]
```diff
@@ -101,16 +101,12 @@ class EasyDownloadServlet(app: EasyDownloadApp) extends ScalatraServlet
...
     Ok()
   }
 
-  private def getLicenseLinkText(uuid: UUID, path: Path): Try[Option[String]] = {
-    for {
-      fileItem <- app.getFileItem(uuid, path)
-      licenseLinkText <- app.configuration.licenses.getLicenseLinkText(fileItem)
-    } yield licenseLinkText
+  private def getLicenseLinkText(fileItem: Try[FileItem]): Option[String] = {
+    fileItem.map(file => Some(s"""<${file.licenseKey}>; rel="license"; title="${file.licenseTitle}"""")).getOrElse(None)
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.download/EasyDownloadServlet.scala]: Am I correct in saying that when `easy-download` calls `easy-auth-info` and the latter returns an error (for any arbitrary reason), this error is not translated into an error response in the former? In other words, it does not cause a `500 Internal Server Error`, `503 Service Unavailable` or something similar?

Comment (vesaakerman) [reply src/main/scala/nl.knaw.dans.easy.download/EasyDownloadServlet.scala]: `fileitem` is fetched in method `getFileItem` in `AuthorisationComponent` and it returns an error if the fileitem can't be fetched.

**Pull request merged**